engine/src/main/java/org/terasology/rendering/nui/editor/utils/NUIEditorMenuTreeBuilder.java

```diff
@@ -250,7 +252,7 @@ private void populateContextMenu(JsonTree node, MenuTree addTree, boolean isSkin
                 // If the node is part of a layout, add an option to add the layoutInfo node.
                 if (nodeInfo.getLayoutClass() != null) {
                     String layoutInfo = "layoutInfo";
-                    if (!node.hasChildWithKey(layoutInfo)) {
+                    if (!Objects.equals(node.getValue().getKey(), layoutInfo) && !node.hasChildWithKey(layoutInfo)) {
```

Comment:
Right, so in this case having a "whitelist" of nodes that can have `layoutInfo` added to them would make more sense. Can you think of any criteria?